contracts/EncodeDAOCore.sol

```diff
@@ -3,16 +3,57 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/access/AccessControl.sol";
 import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
 
 contract EncodeDAOCore is ERC721URIStorage, AccessControl {
+    using Counters for Counters.Counter;
+
+    /// Events
+    event ProposeIssue(
+        address indexed _from,
+        string name,
+        uint256 fundingMinimum,
+        string description
+    );
+
+    /// Constants
     bytes32 public constant MODERATOR_ROLE = keccak256("MODERATOR_ROLE");
 
+    /// State vars
+    Issue[] private currentIssues;
+    Counters.Counter private _issueIds;
+
+    struct Issue {
+        uint256 id;
+        string name;
+        address proposer;
+        uint256 fundingMinimum;
+        string description;
+    }
+
     constructor() ERC721("ApartmentNFT", "ANFT") {
         _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
     }
 
-    /// Propose issue by name and funding
-    function proposeIssue(string memory name, uint256 fundingMinimum) public {}
+    /// Propose issue by name and minimum funding required
+    function proposeIssue(
+        string memory name,
+        uint256 fundingMinimum,
+        string memory description
```

Comment:
Should we have some requirements for how long a description and name can be?

Reply:
Thought of using `bytes` rather than string here as its cheaper, we can then maybe enforce anything within the bytes32 size

Reply:
Yes, something short and sweet haha 😬

Reply:
Maybe description can be string type as it could require some more description on some issues or proposal